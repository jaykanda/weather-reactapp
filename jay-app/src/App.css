.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 50px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
.leftContent {
  float: left;
  width: 15%;
  padding: 15px;
}

.rightContent {
  width: 85%;
  float: left;
}

form {
  float: left;
}

label {
  padding: 5px;
  color: #333;
}

select {
  padding: 10px;
  margin-top: 5px;
  font-size: 14px;
  color: #333;
  clear: both; 
  border-radius: 3px;
}
.selectedCity {
  margin-top: 15px;
}
.selectedCity h5{
  font-size: 16px;
  margin: 0;
  padding: 0;
}
.forecast {
  float: left;
  width: 250px;
  height: 100px;
  border: 1px solid #c2c2c2;
  margin: 15px;
  text-align: center;
  color: #333;
  background-color: #D2E5FF;
  border: 1px solid #B0C0D4;
  border-radius: 3px;
}
.forecast div {
  font-size: 16px;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 500;
  margin: 5px auto;
  text-align: center;
  width: 100%;
}
.forecast div.forecastType {
  font-size: 20px;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 400;
  margin: 5px auto 10px auto;
  text-align: center;
  width: 100%;
}
.forecast span {
  display: block;
  text-align: center;
  font-size: 12px;

}
.weatherReport {
  margin: 0 auto;
  padding: 5px;
  width: 850px;
  border: 1px solid #c2c2c2;
  border-radius: 3px;
  float: left;
  margin: 15px;
}
.cityItem {
  font-style: italic;
}
