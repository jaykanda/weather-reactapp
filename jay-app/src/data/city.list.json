[{
  "id": 707860,
  "name": "Bengaluru",
  "country": "IN",
  "coord": {
    "lon": 34.283333,
    "lat": 44.549999
  }
},
{
  "id": 519188,
  "name": "Hyderabad",
  "country": "IN",
  "coord": {
    "lon": 37.666668,
    "lat": 55.683334
  }
}
]