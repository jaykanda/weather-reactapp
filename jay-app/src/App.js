import React, { Component } from 'react';
import CityDropDown from './component/citydropdown';
import Weatherdashboards from './component/weatherDashboard';
import './App.css';
import * as DATA from './data/city.list.json';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cities : DATA,
      firstCity : 'Bengaluru'
    }   
    this.changeCity = (newCity) => {
      this.setState({firstCity: newCity});
    } 
  }

  render() {    
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Weather report dashboard</h1>
        </header>
        <CityDropDown cityList={this.state.cities} changeCity={this.changeCity}></CityDropDown>
        <Weatherdashboards reqCity={this.state.firstCity}></Weatherdashboards>
      </div>

    );
  }
}

export default App;
