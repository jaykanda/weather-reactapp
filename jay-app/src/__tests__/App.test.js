import React from 'react';
import { shallow, mount, render } from 'enzyme'
import * as WEATHER from '../data/weather-report';

import App from '../App';
import CityDropDown from '../component/citydropdown';
import Weatherdashboards from '../component/weatherDashboard';


const cityList = ['Bengaluru','Hyderabad'];
describe("WEATHER REPORT TEST SUITE", () => {
  it("APP component is loading properly", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.App-title').html()).toMatch('Weather report dashboard');
    expect(wrapper).toMatchSnapshot();  
  });
  it("City dropdown component is loading properly", () => {
    const wrapper = shallow(<CityDropDown cityList={[]} />);
    wrapper.find('.cityItem').forEach((node, i) => {
      expect(node.text()).toEqual(cityList[i]);       
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("Weather report component is loading properly", () => {
    const a = WEATHER;
    const expectItems = ['weather','clouds','wind','snow'];
    const wrapper = shallow(<Weatherdashboards reqCity={cityList[0]}/>);
    wrapper.find('.forecastType').forEach((node) => {
      expect(node.text().toLowerCase().indexOf(expectItems)).toBeGreaterThanOrEqual(-1);      
    });
    expect(wrapper).toMatchSnapshot();
  });
});

