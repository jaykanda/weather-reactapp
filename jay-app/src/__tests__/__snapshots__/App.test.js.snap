// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite APP component is loading properly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Weather report dashboard
    </h1>
  </header>
  <CityDropDown
    changeCity={[Function]}
    cityList={
      Object {
        "0": Object {
          "coord": Object {
            "lat": 44.549999,
            "lon": 34.283333,
          },
          "country": "IN",
          "id": 707860,
          "name": "Bengaluru",
        },
        "1": Object {
          "coord": Object {
            "lat": 55.683334,
            "lon": 37.666668,
          },
          "country": "IN",
          "id": 519188,
          "name": "Hyderabad",
        },
        "default": Array [
          Object {
            "coord": Object {
              "lat": 44.549999,
              "lon": 34.283333,
            },
            "country": "IN",
            "id": 707860,
            "name": "Bengaluru",
          },
          Object {
            "coord": Object {
              "lat": 55.683334,
              "lon": 37.666668,
            },
            "country": "IN",
            "id": 519188,
            "name": "Hyderabad",
          },
        ],
      }
    }
  />
  <Weatherdashboards
    reqCity="Bengaluru"
  />
</div>
`;

exports[`A suite contains spec with an expectation 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Weather report dashboard
    </h1>
  </header>
  <CityDropDown
    changeCity={[Function]}
    cityList={
      Object {
        "0": Object {
          "coord": Object {
            "lat": 44.549999,
            "lon": 34.283333,
          },
          "country": "IN",
          "id": 707860,
          "name": "Bengaluru",
        },
        "1": Object {
          "coord": Object {
            "lat": 55.683334,
            "lon": 37.666668,
          },
          "country": "IN",
          "id": 519188,
          "name": "Hyderabad",
        },
        "default": Array [
          Object {
            "coord": Object {
              "lat": 44.549999,
              "lon": 34.283333,
            },
            "country": "IN",
            "id": 707860,
            "name": "Bengaluru",
          },
          Object {
            "coord": Object {
              "lat": 55.683334,
              "lon": 37.666668,
            },
            "country": "IN",
            "id": 519188,
            "name": "Hyderabad",
          },
        ],
      }
    }
  />
  <Weatherdashboards
    reqCity="Bengaluru"
  />
</div>
`;

exports[`WEATHER REPORT TEST SUITE APP component is loading properly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Weather report dashboard
    </h1>
  </header>
  <CityDropDown
    changeCity={[Function]}
    cityList={
      Object {
        "0": Object {
          "coord": Object {
            "lat": 44.549999,
            "lon": 34.283333,
          },
          "country": "IN",
          "id": 707860,
          "name": "Bengaluru",
        },
        "1": Object {
          "coord": Object {
            "lat": 55.683334,
            "lon": 37.666668,
          },
          "country": "IN",
          "id": 519188,
          "name": "Hyderabad",
        },
        "default": Array [
          Object {
            "coord": Object {
              "lat": 44.549999,
              "lon": 34.283333,
            },
            "country": "IN",
            "id": 707860,
            "name": "Bengaluru",
          },
          Object {
            "coord": Object {
              "lat": 55.683334,
              "lon": 37.666668,
            },
            "country": "IN",
            "id": 519188,
            "name": "Hyderabad",
          },
        ],
      }
    }
  />
  <Weatherdashboards
    reqCity="Bengaluru"
  />
</div>
`;

exports[`WEATHER REPORT TEST SUITE City dropdown component is loading properly 1`] = `
<div
  className="leftContent"
>
  <form>
    <label>
      Please select your city
    </label>
    <select
      onChange={[Function]}
    />
    <div
      className="selectedCity"
    >
      Weather forecast for 
      <h5>
        Bengaluru
      </h5>
    </div>
  </form>
</div>
`;

exports[`WEATHER REPORT TEST SUITE Weather report component is loading properly 1`] = `
<div
  className="weatherReport rightContent"
>
  <div>
    12-07-2018
     
    Bengaluru
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Temperature
    </div>
    <div>
      161.45
       F
      <span>
        Min - 
        159.086
         F
      </span>
      <span>
        Max - 
        161.45
         F
      </span>
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Pressure
    </div>
    <div>
      923.48
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Sea Level
    </div>
    <div>
      945.39
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Ground Level
    </div>
    <div>
      923.48
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Humidity
    </div>
    <div>
      39
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Weather
    </div>
    <div>
      clear sky
    </div>
    <img
      alt="Weather Update"
      src="../images/02n.png"
    />
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Clouds
    </div>
    <div>
      5
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Wind
    </div>
    <div>
      Speed - 
      4.77
       mph
    </div>
    <div>
      Direction - 
      232.505
    </div>
  </div>
  <div
    className="forecast"
  >
    <div
      className="forecastType"
    >
      Snow
    </div>
    <div />
  </div>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Prev 3 Hrs
  </button>
    
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next 3 Hrs
  </button>
</div>
`;
